---
title: "Microbiome DADA2"
author: "Andrew Michael"
date: "2024-04-16"
output: pdf_document
---

# load required packages
```{r}
library(dada2)
```

# load sequences
```{r}
path<-"sequences"
list.files(path)
```

# read in file names
```{r}
# fastq filenames have format: SAMPLENAME_R1_001.fastq
fnFs <- sort(list.files(path, pattern="_R1_001.fastq", full.names = TRUE))
# Extract sample names, assuming filenames have format: SAMPLENAME_XXX.fastq
sample.names <- sapply(strsplit(basename(fnFs), "_"), `[`, 1)
```

# inspect read quality
```{r}
plotQualityProfile(fnFs[1:2])
```

# filter and trim
```{r}
# Place filtered files in filtered/ subdirectory
filtFs <- file.path(path, "filtered", paste0(sample.names, "_F_filt.fastq.gz"))

names(filtFs) <- sample.names

# quality decreases at 120
out <- filterAndTrim(fnFs, filtFs, truncLen=c(120),
              maxN=0, maxEE=c(2), truncQ=2, rm.phix=TRUE,
              compress=TRUE, multithread=TRUE) # On Windows set multithread=FALSE
head(out)
```

# learn error rates
```{r}
errF <- learnErrors(filtFs, multithread=TRUE)
plotErrors(errF, nominalQ=TRUE)
```

# sample inference
```{r}
dadaFs <- dada(filtFs, err=errF, multithread=TRUE)
```

# construct sequence table, this would be important to see if something went wrong in your work, we should see 120 nucleotides, and we do
```{r}
seqtab <- makeSequenceTable(dadaFs)
dim(seqtab)
# Inspect distribution of sequence lengths
table(nchar(getSequences(seqtab)))
```

# remove chimeras, removes the ones that could be the same
```{r}
seqtab.nochim <- removeBimeraDenovo(seqtab, method="consensus", multithread=TRUE, verbose=TRUE)
dim(seqtab.nochim)
```

# track each step through the pipelines, at each step, where did we lose some of the reads
```{r}
getN <- function(x) sum(getUniques(x))
track <- cbind(out, sapply(dadaFs, getN), rowSums(seqtab.nochim))
colnames(track) <- c("input", "filtered", "denoisedF", "nonchim")
rownames(track) <- sample.names
head(track)
```

#save seqtab.nochim as R file
```{r}
save(seqtab.nochim, file="RData/seqtab.nochim.RData")
```

# load seqtab.nochim to start here
```{r}
load("RData/seqtab.nochim.RData")
```

# assign taxonomy
```{r}

